src/modules/api-status/api-status.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CreateApiStatusDto } from './dto/create-api-status.dto';
+import { ApiHealth, ApiStatus } from './entities/api-status.entity';
+import { Request } from './entities/request.entity';
+
+@Injectable()
+export class ApiStatusService {
+  constructor(
+    @InjectRepository(ApiHealth)
+    private readonly apiHealthRepository: Repository<ApiHealth>,
+    @InjectRepository(Request)
+    private readonly requestRepository: Repository<Request>
+  ) {}
+
+  async create(apiHealthDto: CreateApiStatusDto[]) {
+    const apiHealthList = [];
+
+    for (const eachApiStatus of apiHealthDto) {
+      const apiHealth = await this.apiHealthRepository.findOne({
+        where: { api_group: eachApiStatus.api_group },
+      });
+
+      if (!apiHealth) {
+        const apiRequestList = [];
+        const apiHealth = await this.apiHealthRepository.save(eachApiStatus);
+        await Promise.all(
+          eachApiStatus.requests.map(request => {
+            this.requestRepository.save({
```

Comment:
It is not an error, but best practice to improve readability that you use return to explicitly return the array of promises.
github is already flagging your code for not explicitly awaiting the promises.

Reply:
Done, i think.